pre_commit_hooks/end_of_file_fixer.py

```diff
@@ -6,6 +6,26 @@
 from typing import Sequence
 
 
+def detect_eol_equence(file_obj: IO[bytes]) -> bytes:
+    # readline() doesn't work because it doesn't get \r right
+    eol_marker = b''
+    last_was_eol = False
+    while True:
+        next = file_obj.read(1)
```

Comment:
this is going to be extremely slow

Reply:
I don't have one, just i know this isn't acceptable

Reply:
you've got IO[bytes] so no buffering

Reply:
I think that is just the type annotation and doesn't change the buffering of objects returned by `open`. And I run `strace`.

But maybe you have some documentation?